src/psn/mod.rs

```diff
@@ -289,10 +289,15 @@ impl PackageInfo {
 
         if !crate::utils::hash_file(&mut pkg_file, &self.sha1sum, self.hash_whole_file).await? {
             if let Err(e) = pkg_file.set_len(0).await {
-                error!("Failed to set file lenght to 0: {e}");
+                error!("Failed to set file length to 0: {e}");
                 return Err(DownloadError::Tokio(e));
             }
 
+            if let Err(e) = pkg_file.seek(SeekFrom::End(0)).await {
```

Comment:
My brain is not braining much, but if you are trying to go to the beginning of the file, shouldn't this be `SeekFrom::Start`?

Reply:
After setting length of the file to zero `End(0)` and `Start(0)` shouldn't make functional difference and point to the same place if I understand it correctly (the program works the same way) but I changed it to `Start` per suggestion as I agree it is less of a head-scratcher

Reply:
Alright, got it now. Yeah, should be the same, but Start is probably a bit more clear 👍🏼